### PR TITLE
Add missing setData() method for Piecewise functions

### PR DESCRIPTION
**Pull request by bwspenc**
The recent refactoring of the Piecewise functions omitted the definition of `setData()`, which is used in a Mastodon test.  Add that function and a test to ensure that it works.

### CODE REVIEW

**Comment by bwspenc** [comment]
@dschwen FYI, your PR that I merged broke the MASTODON build. I'm not sure how the tests passed on that. The devel branch is not being updated to next because of this issue. We need to merge this PR into the MOOSE next branch, and then submit a PR with a couple of minor changes to MASTODON (which I also have ready) to get the builds clean again.

File: test/tests/functions/hardcoded_piecewise_linear/tests [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  design = 'PiecewiseLinear.md'
+  [hardcoded_piecewise_linear]
+    type = 'CSVDiff'
+    input = 'hardcoded_piecewise_linear.i'
+    csvdiff = 'hardcoded_piecewise_linear_out.csv'
+    requirement = "The Function system shall allow for piecewise functions to directly set the xy data internally"
```
Comment (aeslaughter) [inline test/tests/functions/hardcoded_piecewise_linear/tests]:
```suggestion
    requirement = "The system shall allow for piecewise functions to directly set the x and y data internally."
```

**Review by aeslaughter — DISMISSED** [review]

**Comment by dschwen** [comment]
We may have to make sure that setting the data will recoarsen the function in the Bison class

**Review by aeslaughter — APPROVED** [review]

**Comment by bwspenc** [comment]
@aeslaughter Looks like the tests just passed, so this should be good to merge.

**Pull request merged**